tt_metal/detail/reports/compilation_reporter.cpp

```diff
@@ -79,13 +79,13 @@ std::string kernel_attributes_str(std::shared_ptr<Kernel> kernel) {
     return attr_str;
 }
 
-void CompilationReporter::add_kernel_compile_stats(const Program &program, std::shared_ptr<Kernel> kernel, bool cache_hit, size_t kernel_hash) {
+void CompilationReporter::add_kernel_compile_stats(uint64_t program_id, std::shared_ptr<Kernel> kernel, bool cache_hit, size_t kernel_hash) {
```

Comment:
uint64 for program id?  wonder about how many of these we'll store and if uint32 would be more cache friendly

Reply:
Happy to change that but would prefer to make that change in a separate PR. `Program::get_id()` currently returns `uint64_t` and I wanted to make as few observable changes as necessary to internalize the implementation of Program to the translation unit.